code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt

```diff
@@ -920,22 +925,24 @@ internal class AnalyticsExtension(extensionApi: ExtensionApi) : Extension(extens
         if (analyticsState.isVisitorIDServiceEnabled) {
             analyticsVars.putAll(analyticsState.analyticsIdVisitorParameters)
         }
-        if (ServiceProvider.getInstance().appContextService == null) {
+
+        if (ServiceProvider.getInstance().appContextService != null) {
+            val appState = ServiceProvider.getInstance().appContextService.appState
+            if (appState == AppState.BACKGROUND) {
+                analyticsVars[AnalyticsConstants.ANALYTICS_REQUEST_CUSTOMER_PERSPECTIVE_KEY] =
+                    AnalyticsConstants.APP_STATE_BACKGROUND
+            } else {
+                analyticsVars[AnalyticsConstants.ANALYTICS_REQUEST_CUSTOMER_PERSPECTIVE_KEY] =
+                    AnalyticsConstants.APP_STATE_FOREGROUND
+            }
+        } else {
```

Comment:
1.x seems to default to foreground if the service is null, which should be covering the case where the app is built with fragments and does not manage app state